helix-term/src/commands.rs

```diff
@@ -1541,6 +1560,21 @@ fn delete_selection_impl(reg: &mut Register, doc: &mut Document, view_id: ViewId
     doc.apply(&transaction, view_id);
 }
 
+fn delete_selection_insert_mode(cx: &mut Context) {
+    let (view, doc) = current!(cx.editor);
+    let view_id = view.id;
+    let selection = doc.selection(view_id);
+
+    // then delete
+    let transaction = Transaction::change_by_selection(doc.text(), selection, |range| {
+        (range.from(), range.to(), None)
+    });
+    doc.apply(&transaction, view_id);
+
+    // exit select mode, if currently in select mode
+    exit_select_mode(cx);
```

Comment:
Please just use delete_selection_impl. In insert mode we don't want to do any logic related to exiting select mode.

Reply:
The delete_selection_impl would yank the text to a register, is there a `_` register for this?

Reply:
Hmm, maybe use

```rust
    let transaction = Transaction::change_by_selection(doc.text(), selection, |range| {
        (range.from(), range.to(), None)
    });
    doc.apply(&transaction, view_id);
```

directly in your implementations then?

Reply:
It's still need to make a selection to pass in change_by_selection, right? So I just need remove the `    exit_select_mode(cx);
`?